modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/GroupResourceManagerTest.java

```diff
@@ -747,12 +763,180 @@ public void testUpdateGroupWithPUTNotImplementedException(String id, String scim
                 attributes, excludeAttributes);
         Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_NOT_IMPLEMENTED);
     }
+
+    @DataProvider(name = "dataForUpdateWithPATCH")
+    public Object[][] dataToUpdateWithPATCH() throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+        Group groupNew = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new Group());
+        return new Object[][]{
+                {id, SCIM2_PATCH_REQUEST_STRING, "userName", null, groupNew, groupNew}
+        };
+    }
+
+    @Test(dataProvider = "dataForUpdateWithPATCH")
+    public void testUpdateWithPATCH(String existingId, String patchRequest,
+                                    String attributes, String excludeAttributes, Object scimNewGroupObject,
+                                    Object scimOldGroupObject)
+            throws BadRequestException, CharonException, NotImplementedException, NotFoundException {
+
+        Group groupNew = (Group) scimNewGroupObject;
+        Group groupOld = (Group) scimOldGroupObject;
+        abstractResourceManager.when(()
+                -> AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(SCIM2_GROUP_ENDPOINT);
+        Mockito.when(userManager.getGroup(anyString(), anyMap())).thenReturn(groupOld);
+        Mockito.when(userManager.patchGroup(anyString(), anyString(), anyMap(), anyMap())).thenReturn(groupNew);
+        SCIMResponse scimResponse = groupResourceManager.updateWithPATCH(existingId, patchRequest,
+                userManager, attributes, excludeAttributes);
+        Assert.assertEquals(scimResponse.getResponseStatus(), ResponseCodeConstants.CODE_OK);
+
+    }
+
+    @DataProvider(name = "dataForUpdateWithPATCHOverride")
+    public Object[][] dataToUpdateWithPATCHOverride()
+            throws BadRequestException, CharonException, InternalErrorException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getGroupResourceSchema();
+        JSONDecoder decoder = new JSONDecoder();
+        Group groupOld = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING, schema, new Group());
+        String id = groupOld.getId();
+        Group groupNew = decoder.decodeResource(NEW_GROUP_SCIM_OBJECT_STRING_FOR_PATCH_UPDATE, schema, new Group());
+        return new Object[][]{
+                {id, SCIM2_PATCH_REQUEST_STRING, groupNew, groupNew}
+        };
+    }
+
+    @Test(dataProvider = "dataForUpdateWithPATCHOverride")
+    public void testUpdateWithPATCHOverride(String existingId, String patchRequest,
```

Comment:
what is the behavior that is getting tested here?

Reply:
Update with patch has method override as follows 
``  public SCIMResponse updateWithPATCH(String existingGroupId, String patchRequest, UserManager userManager) ``
and
`` public SCIMResponse updateWithPATCH(String existingId, String patchRequest, UserManager userManager,
                                        String attributes, String excludeAttributes) ``
 This test for the second method definition